lib/bolt/applicator.rb

```diff
@@ -119,17 +128,15 @@ def compile(target, catalog_input)
         { 'level' => 'err', 'message' => line }
       end
 
-      result = JSON.parse(out)
-      if stat.success?
+      if stat.success? || print_error
```

Comment:
Since we have this variable anyway, we may as well use it as the sole condition variable. We can set `print_error = stat.success?` at the top and override it in the `rescue` if necessary when parsing results.

Maybe `print_logs` would be better because these aren't necessarily errors? That also makes this code read nicely: `if print_logs [... print the logs ...]`